clients/web/test/testUtils.js

```diff
@@ -11,6 +11,17 @@ window.alert = function (msg) {
     expect('Alert was used').toBe('Alerts should not be present');
 };
 
+// Augment jQuery's click() function to follow href properties in links if they exist.
+$.fn.click = _.wrap($.fn.click, function (old) {
+    var args = Array.prototype.slice.call(arguments, 1);
+    var href = $(this).attr('href');
```

Comment:
You only really need to bother building and evaluating this selector if `args.length === 0`, so you should probably have a nested `if` (rather than a single level with combined conditions) and create this selector only if necessary.

Reply:
Do you need to re-wrap `this`? We're already calling `.click()` on a jQuery selector, right?

Reply:
While uncommon, it is possible for non-`<a>` elements to have an `href` attribute, sometimes with different semantics. I think we should also check that `this` is an `<a>` element.

Reply:
Probably a good idea, although I'm curious what other elements support href attributes?

Reply:
From a quick search of MDN: several different things under `svg`, `area`, etc.

When we're monkey-patching such a core part of jQuery in our test environment, it's probably best to not make assumptions about what's being clicked.